internal/pkg/services/argus/utils/utils.go

```diff
@@ -56,6 +57,143 @@ func LoadPlanId(planName string, resp *argus.PlansResponse) (*string, error) {
 	}
 }
 
+func MapToUpdateScrapeConfigPayload(resp *argus.GetScrapeConfigResponse) (*argus.UpdateScrapeConfigPayload, error) {
+	if resp == nil || resp.Data == nil {
+		return nil, fmt.Errorf("no Argus scrape config provided")
+	}
+
+	data := resp.Data
+
+	basicAuth := mapBasicAuth(data.BasicAuth)
+
+	var staticConfigs *[]argus.UpdateScrapeConfigPayloadStaticConfigsInner
+	if data.StaticConfigs != nil {
+		configs := make([]argus.UpdateScrapeConfigPayloadStaticConfigsInner, 0)
+		for _, config := range *data.StaticConfigs {
+			newConfig, err := mapStaticConfig(config)
+			if err != nil {
+				return nil, fmt.Errorf("map static config: %w", err)
+			}
+			configs = append(configs, newConfig)
+		}
+		staticConfigs = &configs
+	}
```

Comment:
I suggest some changes to this block (and the following ones) to make it more readable:

```
var staticConfigs []argus.UpdateScrapeConfigPayloadStaticConfigsInner
if data.StaticConfigs != nil {
    for _, config := range *data.StaticConfigs {
        newConfig, err := mapStaticConfig(config)
        if err != nil {
            return nil, fmt.Errorf("map static config: %w", err)
        }
        staticConfigs = append(staticConfigs, newConfig)
    }
}

(...)

return &argus.UpdateScrapeConfigPayload{
		BasicAuth:             basicAuth,
		BearerToken:           data.BearerToken,
		HonorLabels:           data.HonorLabels,
		HonorTimeStamps:       data.HonorTimeStamps,
		MetricsPath:           data.MetricsPath,
		(...)
		StaticConfigs:         &staticConfigs,
		(...)
	}, nil
```

Doing `var staticConfigs []argus.UpdateScrapeConfigPayloadStaticConfigsInner` or `make([]argus.UpdateScrapeConfigPayloadStaticConfigsInner, 0)` is the same thing and the first one is more readable. What you could do instead would be `make([]argus.UpdateScrapeConfigPayloadStaticConfigsInner, len(*data.StaticConfigs))` but then this would require the nil check first which would make it more complex without much added improvement on performance for most likely small arrays

Reply:
I actually was doing it this way, the problem with this is that the field in the API Request is optional so it should be set to nil if not used right? 
Because this way we always set a pointer to something in the API request, even if it is a pointer to an empty array. But if setting to nil is not a requirement, then this is much cleaner!

Reply:
Yeah, I agree. The original thinking was that the way you suggest I would always return a pointer to a list instead of a nil pointer if it is not set in the initial payload.
I refactored the code overall, should also fix this problem.

Reply:
The zero value of a slice is `nil`.
So if there is nothing on the initial payload, we will have a pointer to `nil`, which is the same as a `nil` pointer